src/google-classroom-extractor/edfi_google_classroom_extractor/api/courses.py

```diff
@@ -20,6 +20,178 @@
 logger = logging.getLogger(__name__)
 
 
+def request_course_aliases(
+    resource: Optional[Resource], course_id: str
+) -> List[Dict[str, str]]:
+    """
+    Fetch CourseAlias API data for the given course id
+
+    Parameters
+    ----------
+    resource: Optional[Resource]
+        a Google Classroom SDK Resource
+    course_id: str
+        a Google Classroom course id
+
+    Returns
+    -------
+    List[Dict[str, str]]
+        a list of Google Classroom CourseAliases for the given course id,
+            see https://developers.google.com/classroom/reference/rest/v1/courses.aliases
+    """
+
+    if resource is None:
+        return []
+
+    return call_api(
+        cast(ResourceType, resource).courses().aliases().list,
+        {"courseId": course_id},
+        "aliases",
+    )
+
+
+def _alias_names_for_scope(aliases: List[str], scope: str) -> List[str]:
+    """
+    Returns a list of alias names from a list of aliases with the given scope.
+    Aliases from the API are scoped with a prefix character followed by a colon
+    and then the alias name. For example "d:alias_name_1".
+
+    Parameters
+    ----------
+    aliases: List[str]
+        a list of course alias strings from the API
+    scope: str
+        the scope to filter by
+
+    Returns
+    -------
+    List[str]
+        a list of alias names (scope omitted) filtered by the given scope
+    """
+    result: List[str] = []
+    try:
+        result = [
+            alias.split(":")[1] for alias in aliases if alias.split(":")[0] == scope
+        ]
+    except (ValueError, IndexError):
+        pass
```

Comment:
Any value to having a unit tests for the `except` case? I think the main thing would be insuring that the calling function handles the empty list.

Reply:
Added test for malformed aliases.